components/global/product_switcher_menu/product_switcher_menu.tsx

```diff
@@ -45,27 +46,18 @@ type Props = {
 // TODO: rewrite this to a functional component
 class ProductSwitcherMenu extends React.PureComponent<Props> {
     static defaultProps = {
-        teamType: '',
         isMobile: false,
         pluginMenuItems: [],
     };
 
-    handleEmitUserLoggedOutEvent = () => {
-        GlobalActions.emitUserLoggedOutEvent();
-    }
-
-    render() {
-        const {currentUser, isMessaging, isMobile, onClick} = this.props;
+    handleEmitUserLoggedOutEvent = (): void => GlobalActions.emitUserLoggedOutEvent();
 
-        if (!currentUser) {
-            return null;
-        }
+    render(): JSX.Element {
+        const {appDownloadLink, isMessaging, isMobile, onClick, intl: {formatMessage}} = this.props;
```

Comment:
If we are going to de-structure we should at least be consistent.  I would prefer leaving everything as `this.props.*` though.

Reply:
IMO it needs to be decided on a per case basis: the rule of thumb for my decision-making on this is always: "If you access the property more than once from `this.props.*` it is better to destructure it." I am open for a discussion about this.

Reply:
I've gone back and forth on destructuring everything in the past. I used to be against destructuring entirely since I like verbosity and knowing where stuff comes from, but I've definitely changed my opinion over time since that's not always relevant (unless you're doing something like comparing `props` to `nextProps` where it's still relevant because there's 2 objects with the same fields).

I do tend to prefer going all or nothing on destructuring whenever possible, but that's just me though.

Reply:
It is definitely easier when using functional components, where you typically destructure everything from within the parameter list itself. As for "old-school" components ... well, this is again very much question of personal preference and/or global style-decisions.